adal/src/main/java/com/microsoft/aad/adal/Oauth2.java

```diff
@@ -498,10 +508,12 @@ private AuthenticationResult postMessage(String requestMessage, Map<String, Stri
                     }
 
                     if (mRequest.getIsExtendedLifetimeEnabled()) {
+                        stopHttpEventWithFailure(httpEvent);
                         Logger.v(TAG, "WebResponse is not a success due to: " + response.getStatusCode());
                         throw e;
                     } else {
                         Logger.v(TAG, "WebResponse is not a success due to: " + response.getStatusCode());
+                        stopHttpEventWithFailure(httpEvent);
```

Comment:
> stopHttpEventWithFailure(httpEvent); [](start = 23, length = 37)

Instead of having the stopevent everywhere, can we just wrap it around the sendPost? status code is only thing we care for HttpEvent, right?